### PR TITLE
Oj 2337 process event to add to txma queue

### PR DESCRIPTION
**Pull request by Connor-Johnson-DCO**
### What changed

- EventBridge rules
    - Now enabled in dev environments only
    - Input transformation create minimum required data for TXMA event
    - Target the audit event queue (TXMA)
- Updated current time lambda to return seconds and milliseconds
- State machines now use current time lambda
- Refactored the flow of data in issue credential lambda 
- Fixed tests broken by changes
- Updated the data returned by check session state machine

### Why did it change

- Allow events to be triggered from state machines 
- Events go onto the audit event queue for TXMA

### Issue tracking

- [OJ-2337](https://govukverify.atlassian.net/browse/OJ-2337)

### Environment variables or secrets

- [ ] No environment variables or secrets were added or changed

- [ ] Documented in the [README](./blob/main/README.md)
- [ ] Added to deployment repository
- [ ] Added to local startup repository

### Other considerations

- [ ] Update [README](./blob/main/README.md) with any new instructions or tasks

[OJ-2337]: https://govukverify.atlassian.net/browse/OJ-2337?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: infrastructure/template.yaml [hunk 1]
```diff
@@ -1056,60 +1065,225 @@ Resources:
     Properties:
       EventBusName: !Ref CheckHmrcEventBus
       Description: Capture the audit event RESPONSE_RECEIVED
-      State: DISABLED
+      State: !If [IsDevLikeEnvironment, ENABLED, DISABLED]
```
Comment (Caitlin-cooling) [inline infrastructure/template.yaml:1068]: What's the reason we only want these in dev?

File: lambdas/matching/src/current-time-handler.ts [hunk 2]
```diff
@@ -1,8 +1,13 @@
 import { LambdaInterface } from "@aws-lambda-powertools/commons";
 
 export class CurrentTimeHandler implements LambdaInterface {
-  public async handler(_event: unknown, _context: unknown): Promise<string> {
-    return Math.floor(Date.now() / 1000).toString();
+  public async handler(_event: unknown, _context: unknown): Promise<Object> {
```
Comment (Caitlin-cooling) [inline lambdas/matching/src/current-time-handler.ts:4]: Do we have any unit tests for this handler?

File: step-functions/nino_issue_credential.asl.json [hunk 3]
```diff
@@ -346,7 +391,10 @@
         "Entries": [
           {
             "Detail": {
-              "message": "VC Successfully Issued"
+              "msTimestamp.$": "$.currentTime.milliseconds",
+              "sTimestamp.$": "$.currentTime.seconds",
+              "session.$": "$.session",
+              "issuer.$": "$.parameters.issuer"
```
Comment (Caitlin-cooling) [inline step-functions/nino_issue_credential.asl.json:397]: In other VC issued events I've seen we've included `ci` and `ciReasons` if there are any. I can't see any payloads on the tickets but I wondered if we wanted to do that here? Maybe it's a question for Sanka?

File: step-functions/nino_check.asl.json [hunk 4]
```diff
@@ -176,14 +177,30 @@
         "value.$": "$.SecretString"
       }
     },
+    "Fetch Current Time For Request Sent": {
```
Comment (surajkumar) [inline step-functions/nino_check.asl.json:180]: Let's refactor this a little bit. We can modify the "Call Matching API" to return the start and end times needed for the TxMA events. This removes the need for the additional 2 Lambda calls to get the time. 

As TxMA events are asynchronous we don't need to do them 1 by 1. Instead, the `EventBridge: PutEvents` actually accepts an array of events https://docs.aws.amazon.com/eventbridge/latest/APIReference/API_PutEvents.html so the TxMA events for nino-check can be done in 1 single state.

In return, this means we only added 1 state to the state machine, keeping the complexity of multiple states/lambdas out.

File: step-functions/nino_issue_credential.asl.json [hunk 5]
```diff
@@ -333,11 +365,24 @@
           "BackoffRate": 2
         }
       ],
+      "Next": "Fetch Current Time For VC Issued Event",
+      "ResultSelector": {
+        "value.$": "$.Payload"
+      },
+      "ResultPath": "$.vcClaimSetSignature"
+    },
+    "Fetch Current Time For VC Issued Event": {
```
Comment (surajkumar) [inline step-functions/nino_issue_credential.asl.json:374]: Similar to my previous comment, in this instance, you are sending the TxMA events both at the same time. Any reason it needs 2 different timestamps? I think we can leave it as 1 lambda to get the time and 1 `EventBridge: PutEvents` to send all the TxMA events.

**Review by surajkumar — CHANGES_REQUESTED** [review]
Let's have a group discussion on Monday to address the changes and existing comments on this PR.

File: step-functions/nino_issue_credential.asl.json (hunk 3 above)
Comment (Connor-Johnson-DCO) [reply step-functions/nino_issue_credential.asl.json:397]: The main goal of this ticket was to setup the infrastructure, and at the moment the payload for the events just cover all of the base properties required for TXMA. Follow up tickets will add further properties when we have had discussions with TXMA on exactly what they want for each event.

File: step-functions/nino_issue_credential.asl.json (hunk 5 above)
Comment (Connor-Johnson-DCO) [reply step-functions/nino_issue_credential.asl.json:374]: I don't know how important precision of these events are. When looking at the timestamps after doing a run through, they are different but only by milliseconds.

File: infrastructure/template.yaml (hunk 1 above)
Comment (Connor-Johnson-DCO) [reply infrastructure/template.yaml:1068]: This was part of the acceptance criteria, I think this was due to the fact that the properties inside the current events are not yet final. A further talk with TXMA to get exact details are required.

**Pull request closed without merging**